inkcpp/include/functional.h

```diff
@@ -59,7 +59,7 @@ namespace ink::runtime::internal
 
 		// pops an argument from the stack using the function-type
 		template<int index>
-		typename arg_type<index> pop_arg(basic_eval_stack* stack)
+		arg_type<index> pop_arg(basic_eval_stack* stack)
```

Comment:
`typename` can only be for going into something (like `foo<int>::bar`), in this case it can only be a type, so it's forbidden to write it.